packages/eslint-plugin/src/rules/prefer-signal-readonly.ts

```diff
@@ -0,0 +1,105 @@
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+type Options = [];
+export type MessageIds = 'preferSignalReadonly' | 'suggestAddReadonlyModifier';
+export const RULE_NAME = 'prefer-signal-readonly';
+const KNOWN_SIGNAL_TYPES: ReadonlySet<string> = new Set([
+  'InputSignal',
+  'ModelSignal',
+  'Signal',
+  'WritableSignal',
+]);
+const KNOWN_SIGNAL_CREATION_FUNCTIONS: ReadonlySet<string> = new Set([
```

Comment:
I think we should check the type of the signal instead of the function name. There are many custom utility functions that return signals.

Reply:
Type-aware linting used to have a significant performance cost, is this still the case in newer versions? If so, having a faster but more limited rule may be more desirable for some people.

One way to solve this would probably be to make using types opt-in, but I got no idea if that's something that can be done inside a single rule.

Reply:
A lot of people already have type-aware linting, so I don't think that one more rule will greatly affect performance.

Adding an option to disable it is a good idea, I'll give it a try.

Reply:
https://github.com/reduckted/angular-eslint/pull/1